src/MainWindow.cpp

```diff
@@ -0,0 +1,936 @@
+}
+
+void MainWindow::createCorruptConfigFileBox()
+{
+    // Create the corrupt config file QMessageBox with appropriate displayed text
+    QMessageBox corruptBox;
+    corruptBox.setText(m_CONFIG_CORRUPT_FILE_BOX_PRIMARY_TEXT);
+    corruptBox.setInformativeText(m_CONFIG_CORRUPT_FILE_BOX_INFO_TEXT);
+
+    // Give the user the options of "Ok'ing" the error or closing the application
+    corruptBox.setStandardButtons(QMessageBox::Ok | QMessageBox::Close);
+
+    // Store the user's selection
+    int corruptBoxSelection = corruptBox.exec();
+
+    switch(corruptBoxSelection)
+    {
+        // If the user selected Ok
+        case QMessageBox::Ok :
+        {
+            // Kick off the welcome sequence to have the user re-enter their financial information to get back to a functional state
+            welcomeFirstTimeUser();
+            break;
+        }
+
+            // If the user selected Close
+        case QMessageBox::Close :
+        {
+            // Terminate the application
+            emit conditionToTerminateMet();
+            break;
+        }
+
+        default:
+        {
+            // Default to have the user re-enter their information
+            welcomeFirstTimeUser();
+            break;
+        }
+    }
+}
+
+QString MainWindow::fundingStatusBooleanToString(bool p_isBillFunded)
+{
+    // If the bill has been funded, return the funded string, otherwise return the not funded string
+    return p_isBillFunded ? m_FUNDED_STRING : m_NOT_FUNDED_STRING;
+}
+
+bool MainWindow::fundingStatusStringToBoolean(QString p_fundingStatus)
+{
+    // If the bill has a funding status of "Funded", return true, otherwise return false
+    return p_fundingStatus == m_FUNDED_STRING ? true : false;
+}
+
+QString MainWindow::removeSpaces(QString p_stringWithSpaces)
+{
+    // Convert the input string to a format in which characters can be replaced
+    QString noSpaces = p_stringWithSpaces.simplified();
+
+    // Return the string with spaces removed
+    return noSpaces.replace(" ", "");
+}
+
+void MainWindow::openConfigForBillCreation()
+{
+    // Attempt to access the config file
+    QSettings m_settings(m_CONFIG_FILE_DIRECTORY, QSettings::IniFormat);
+
+    // If the attempt to open the config file results in an error
+    if(m_settings.status() != QSettings::NoError)
+    {
+        // Alert the user with a message box
+        createCorruptConfigFileBox();
+    }
+
+    // Otherwise if the config file was opened successfully
+    else
+    {
+        // Write the Bill information out to the config file
+        m_settings.beginGroup(removeSpaces(m_billWidget->getNameInput()->text()));
+        m_settings.setValue(m_BILL_AMOUNT_DUE_KEY, QString::number(m_billWidget->getAmountDueInput()->text().toDouble(), 'f', 2));
+        m_settings.setValue(m_BILL_DUE_DATE_KEY, m_billWidget->getDueDateInput()->date().toString(m_DATE_STRING_FORMAT));
+        m_settings.setValue(m_BILL_FUNDING_STATUS_KEY, fundingStatusBooleanToString(false));
```

Comment:
Leave a comment mentioning that new bills being written to the config file are defaulted to have a funding status of not funded.

Reply:
Comment updated.